### PR TITLE
Add a compact list of GMT modules

### PR DESCRIPTION
**Pull request by seisman**
Closes https://github.com/gmt-china/GMT_docs/issues/366

Preview:

image

### CODE REVIEW

File: source/module/index.rst [hunk 1]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmtdefaults`
+    - :doc:`gmtget`
+    - :doc:`gmtinfo`
+    - :doc:`gmtlogo`
+    - :doc:`gmtselect`
+    - :doc:`gmtset`
+    - :doc:`gmtsimplify`
+    - :doc:`gmtwhich`
+    - :doc:`grd2cpt`
+    - :doc:`grd2xyz`
```
Comment (core-man) [inline source/module/index.rst:40]: repeated.
```suggestion
```

File: source/module/index.rst [hunk 2]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`grd2xyz`
+    - :doc:`grdblend`
+    - :doc:`grdclip`
+    - :doc:`grdcontour`
+    - :doc:`grdconvert`
```
Comment (core-man) [inline source/module/index.rst:45]: Repeated.
```suggestion
```

File: source/module/index.rst [hunk 3]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`grdconvert`
+    - :doc:`grdcut`
+    - :doc:`grdedit`
+    - :doc:`grdfill`
+    - :doc:`grdimage`
+    - :doc:`grdinfo`
```
Comment (core-man) [inline source/module/index.rst:51]: Repeated
```suggestion
```

File: source/module/index.rst [hunk 4]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`grdinfo`
+    - :doc:`grdlandmask`
+    - :doc:`grdmask`
+    - :doc:`grdpaste`
+    - :doc:`grdproject`
+    - :doc:`grdsample`
+    - :doc:`grdtrend`
```
Comment (core-man) [inline source/module/index.rst:58]: Added `grdtrack`
```suggestion
    - :doc:`grdtrack`
    - :doc:`grdtrend`
```

File: source/module/index.rst [hunk 5]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`text`
+    - :doc:`velo`
+    - :doc:`xyz2grd`
+
+**GMT 模块（尚未翻译整理，欢迎贡献）**
+
+.. hlist::
+    :columns: 6
+
+    - :doc:`gmt:blockmean`
```
Comment (core-man) [inline source/module/index.rst:91]:
```suggestion
    - :doc:`gmt:batch`
    - :doc:`gmt:blockmean`
```

File: source/module/index.rst [hunk 6]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:blockmedian`
+    - :doc:`gmt:blockmode`
+    - :doc:`gmt:dimfilter`
+    - :doc:`gmt:events`
+    - :doc:`gmt:gmt2kml`
```
Comment (core-man) [inline source/module/index.rst:96]:
```suggestion
    - :doc:`gmt:gmt2kml`
    - :doc:`gmtbinstats`
```

File: source/module/index.rst [hunk 7]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:gmtconvert`
+    - :doc:`gmt:gmtmath`
+    - :doc:`gmt:gmtregress`
+    - :doc:`gmt:gmtspatial`
+    - :doc:`gmt:gmtsplit`
+    - :doc:`gmt:gmtswitch`
+    - :doc:`gmt:gmtvector`
+    - :doc:`gmt:grdfft`
```
Comment (core-man) [inline source/module/index.rst:104]:
```suggestion
    - :doc:`gmt:grdfft`
    - :doc:`gmt:grdgdal`
```

File: source/module/index.rst [hunk 8]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:dimfilter`
+    - :doc:`gmt:events`
+    - :doc:`gmt:gmt2kml`
+    - :doc:`gmt:gmtconvert`
+    - :doc:`gmt:gmtmath`
+    - :doc:`gmt:gmtregress`
+    - :doc:`gmt:gmtspatial`
+    - :doc:`gmt:gmtsplit`
+    - :doc:`gmt:gmtswitch`
+    - :doc:`gmt:gmtvector`
```
Comment (core-man) [inline source/module/index.rst:103]:
```suggestion
    - :doc:`gmt:gmtvector`
    - :doc:`gmt:grd2kml`
```

File: source/module/index.rst [hunk 9]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:grdfft`
+    - :doc:`gmt:grdfilter`
+    - :doc:`gmt:grdgradient`
+    - :doc:`gmt:grdhisteq`
```
Comment (core-man) [inline source/module/index.rst:107]:
```suggestion
    - :doc:`gmt:grdhisteq`
    - :doc:`gmt:grdinterpolate`
```

File: source/module/index.rst [hunk 10]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:grdmath`
```
Comment (core-man) [inline source/module/index.rst:108]:
```suggestion
    - :doc:`gmt:grdmath`
    - :doc:`gmt:grdmix`
```

File: source/module/index.rst [hunk 11]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:supplements/segy/segy2grd`
+    - :doc:`gmt:supplements/segy/segy`
+    - :doc:`gmt:supplements/segy/segyz`
+    - :doc:`gmt:supplements/spotter/backtracker`
+    - :doc:`gmt:supplements/spotter/gmtpmodeler`
+    - :doc:`gmt:supplements/spotter/grdpmodeler`
+    - :doc:`gmt:supplements/spotter/grdrotater`
+    - :doc:`gmt:supplements/spotter/grdspotter`
+    - :doc:`gmt:supplements/spotter/hotspotter`
+    - :doc:`gmt:supplements/spotter/originater`
```
Comment (core-man) [inline source/module/index.rst:150]:
```suggestion
    - :doc:`gmt:supplements/spotter/originater`
    - :doc:`gmt:supplements/spotter/polespotter`
```

File: source/module/index.rst [hunk 12]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:greenspline`
+    - :doc:`gmt:mask`
+    - :doc:`gmt:movie`
+    - :doc:`gmt:nearneighbor`
+    - :doc:`gmt:plot3d`
+    - :doc:`gmt:sph2grd`
+    - :doc:`gmt:sphdistance`
+    - :doc:`gmt:sphinterpolate`
+    - :doc:`gmt:sphtriangulate`
+    - :doc:`gmt:supplements/geodesy/gpsgridder`
```
Comment (core-man) [inline source/module/index.rst:119]:
```suggestion
    - :doc:`gmt:supplements/geodesy/earthtide`
    - :doc:`gmt:supplements/geodesy/gpsgridder`
```

**Comment by core-man** [comment]
Do we want to add `gmt5syntax` and `gmt_shell_functions.sh`, which are listed as `Utility Scripts` in GMT documentation?

File: source/module/index.rst [hunk 13]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:supplements/x2sys/x2sys_merge`
+    - :doc:`gmt:supplements/x2sys/x2sys_put`
+    - :doc:`gmt:supplements/x2sys/x2sys_report`
+    - :doc:`gmt:supplements/x2sys/x2sys_solve`
+    - :doc:`gmt:trend1d`
+    - :doc:`gmt:trend2d`
+    - :doc:`gmt:triangulate`
+    - :doc:`gmt:wiggle`
+
+**GMT 模块（按功能分类）**
```
Comment (core-man) [inline source/module/index.rst:168]:
```suggestion
**GMT 模块（按功能分类）[包含了尚未翻译模块]**
```

File: source/module/index.rst [hunk 14]
```diff
@@ -1,6 +1,172 @@
...
+
```
Comment (core-man) [inline source/module/index.rst:169]: Do we want to update the following tables based on https://docs.generic-mapping-tools.org/latest/modules.html#session-management in another PR later?

File: source/module/index.rst [hunk 15]
```diff
@@ -1,6 +1,172 @@
...
+    - :doc:`gmt:movie`
+    - :doc:`gmt:nearneighbor`
+    - :doc:`gmt:plot3d`
+    - :doc:`gmt:sph2grd`
+    - :doc:`gmt:sphdistance`
+    - :doc:`gmt:sphinterpolate`
+    - :doc:`gmt:sphtriangulate`
+    - :doc:`gmt:supplements/geodesy/gpsgridder`
+    - :doc:`gmt:supplements/gshhg/gshhg`
+    - :doc:`gmt:supplements/img/img2google`
```
Comment (core-man) [inline source/module/index.rst:121]: `img2google` is a module? It seems like util. See https://docs.generic-mapping-tools.org/latest/supplements/img/img2google.html.

Comment (seisman) [reply source/module/index.rst:126]: 是的，但是不想把img2google单独提出来

File: source/module/index.rst (hunk 14 above)
Comment (seisman) [reply source/module/index.rst:175]: 这个表太复杂了，应该会改写

File: source/module/index.rst [hunk 16]
```diff
@@ -1,6 +1,178 @@
...
+    :columns: 6
+
+    - :doc:`gmt:batch`
+    - :doc:`gmt:blockmean`
+    - :doc:`gmt:blockmedian`
+    - :doc:`gmt:blockmode`
+    - :doc:`gmt:dimfilter`
+    - :doc:`gmt:events`
+    - :doc:`gmt:gmt2kml`
+    - :doc:`gmtbinstats`
```
Comment (seisman) [inline source/module/index.rst:96]:
```suggestion
    - :doc:`gmt:gmtbinstats`
```

File: source/module/index.rst (hunk 14 above)
Comment (core-man) [reply source/module/index.rst:175]: 跟 GMT 官方一致呢？或许回头我可以慢慢做

File: source/module/index.rst (hunk 15 above)
Comment (core-man) [reply source/module/index.rst:126]: 可以，我提交了PR到GMT官方文档 (https://github.com/GenericMappingTools/gmt/pull/5366)

File: source/module/index.rst [hunk 17]
```diff
@@ -1,6 +1,178 @@
...
+    - :doc:`gmt:grd2kml`
+    - :doc:`gmt:grdfft`
+    - :doc:`gmt:grdgdal`
+    - :doc:`gmt:grdfilter`
+    - :doc:`gmt:grdgradient`
+    - :doc:`gmt:grdhisteq`
+    - :doc:`gmt:grdinterpolate`
+    - :doc:`gmt:grdmath`
+    - :doc:`gmt:grdmix`
+    - :doc:`gmt:grdtrack`
```
Comment (core-man) [inline source/module/index.rst:113]:
```suggestion
```

**Comment by core-man** [comment]
GMT has 1 program (`gmt`), 97 core modules, 50 supplemental modules, and 5 utility scripts. So it's a total of 153.

Now we have 68 completed commands (66 modules + 2 program/scripts) and 83 todos, which is a total of 151. We don't count `gmt5syntax` and `gmt_shell_functions.sh`, so I think we've done.

**Review by core-man — APPROVED** [review]

**Pull request merged**